src/aks-preview/azext_aks_preview/custom.py

```diff
@@ -3274,17 +3414,29 @@ def aks_disable_addons(cmd, client, resource_group_name, name, addons, no_wait=F
 
 def aks_enable_addons(cmd, client, resource_group_name, name, addons, workspace_resource_id=None,
                       subnet_name=None, appgw_name=None, appgw_subnet_prefix=None, appgw_subnet_cidr=None, appgw_id=None, appgw_subnet_id=None,
-                      appgw_watch_namespace=None, enable_sgxquotehelper=False, enable_secret_rotation=False, no_wait=False):
+                      appgw_watch_namespace=None, enable_sgxquotehelper=False, enable_secret_rotation=False, no_wait=False, enable_msi_auth_for_monitoring=False):
+
     instance = client.get(resource_group_name, name)
     subscription_id = get_subscription_id(cmd.cli_ctx)
     instance = _update_addons(cmd, instance, subscription_id, resource_group_name, name, addons, enable=True,
-                              workspace_resource_id=workspace_resource_id, subnet_name=subnet_name,
+                              workspace_resource_id=workspace_resource_id, enable_msi_auth_for_monitoring=enable_msi_auth_for_monitoring, subnet_name=subnet_name,
                               appgw_name=appgw_name, appgw_subnet_prefix=appgw_subnet_prefix, appgw_subnet_cidr=appgw_subnet_cidr, appgw_id=appgw_id, appgw_subnet_id=appgw_subnet_id, appgw_watch_namespace=appgw_watch_namespace,
                               enable_sgxquotehelper=enable_sgxquotehelper, enable_secret_rotation=enable_secret_rotation, no_wait=no_wait)
 
     if CONST_MONITORING_ADDON_NAME in instance.addon_profiles and instance.addon_profiles[CONST_MONITORING_ADDON_NAME].enabled:
-        _ensure_container_insights_for_monitoring(
-            cmd, instance.addon_profiles[CONST_MONITORING_ADDON_NAME])
+        if instance.addon_profiles[CONST_MONITORING_ADDON_NAME].config[CONST_MONITORING_USING_AAD_MSI_AUTH]:
+            # Ensure the cluster does not use service principal auth
+            try:
+                if instance.servicePrincipalProfile.clientId != "":  # just checking if this field exists
```

Comment:
Not sure if this is the best way to check if a cluster uses service principal auth. Please let me know if there is a better way.

Reply:
I don't know much about service principal auth. @jiasli Could you help to answer this question?

Reply:
This doesn't seem to be related to CLI's authentication. I have zero knowledge about AKS instance. 😥

Reply:
Hi, @daweim0 , if the cluster using managed identity, clientId will be "msi" - https://docs.microsoft.com/en-us/azure/aks/use-managed-identity#obtain-and-use-the-system-assigned-managed-identity-for-your-aks-cluster. I think, we should get validated with Tongyao for both SAI & UAI.

sshadmin@gangams-ci-2:~$ az aks show -g gangams-aks-cpu-throttle-test -n gangams-aks-cpu-throttle-test --query "servicePrincipalProfile"
{
  "clientId": "msi"
}

Reply:
That works perfectly, thanks

Reply:
Hi, @daweim0 , as discussed, this needs to be fixed. The fix should be **if instance.servicePrincipalProfile.clientId != "msi"**